### DISCUSSION TITLE
newsboat: cannot open firefox

### DISCUSSION THREAD

**Discussion by githlp** [netblue30/firejail#5287] [Q&A]
my newsboat and firefox work fine jailed, just not together.

when i call newsboat and the want to open an url via the newsboat-browser.sh it claims:

`/home/ronald/.config/newsboat/newsboat-browser.sh: line 2: /usr/bin/firefox: file or dir not found`

newsboat.local
`whitelist ${HOME}/.newsboat`
`noblacklist ${HOME}/.newsboat`
`ignore noexec `
`ignore private-cache`

**Comment by ghost** [comment]
> my newsboat and firefox work fine jailed, just not together.

Our newsboat profile is designed to use a CLI web browser like lynx or w3m. Have a look at /etc/firejail/newsboat.profile, especially the `private-bin` line. Because firefox isn't part of private-bin you get the error you showed above. Now you might think okay, let's add firefox to private-bin and go ahead. There's more secure ways to use firefox with newsboat: see #3894.

> newsboat.local
whitelist ${HOME}/.newsboat
noblacklist ${HOME}/.newsboat
ignore noexec
ignore private-cache

Why do you have these options in there? The first two are already part of /etc/firejail/newsboat.profile and can be removed. Can you elaborate on why you need to ignore noexec and private-cache?

**Comment by githlp** [comment]
Thank you, my bad, the newsboat.local was just a blind shot.

It now worked with the xdg-open version from #3894.

As i read not carefully enough i just put the

`#!/bin/sh
systemd-run --user --quiet --no-block /usr/bin/xdg-open "$@"`

in my newsboat-browser.sh.

Why should i create a file "/usr/local/bin/xdg-open" for it ?

I found a more or less short documentation about general profile creating, but for the moment i am still without a concrete plan about how to create exceptions to the given standard profiles.. e.g. i was not able to figure out why my browsers cannot open the download dir after i downloaded a file, the Download dir itself is accassable but i guess they cannot hand it over to thunar.

**Reply by rusty-snake** [reply]
> i was not able to figure out why my browsers cannot open the download dir after i downloaded a file, the Download dir itself is accassable but i guess they cannot hand it over to thunar.

Yes, that's a feature. Sandboxing means isolating a program from the rest of the system, this includes IPC. For this functionality you need to allow `org.freedesktop.FileManager1`, which (maybe) can be used to escape the sandbox.

**Reply by rusty-snake** [reply]
> Why should i create a file "/usr/local/bin/xdg-open" for it ?

Because `/usr/local/bin` has a higher priority in `$PATH` than `/usr/bin`, so if you place it there you can replace the `xdg-open` implementation that is found and used by programs.

**Reply by githlp** [reply]
well, i was too fast with my guess that it works now.. i had the alias to /usr/bin/newsboat ..

Actually, i still get
`sh: line 1: /home/ronald/.config/newsboat/newsboat-browser.sh: permission denied`

I have my newsboat.local and my /usr/local/bin/xdg-open exactly as in #3894.

Do i need to alter the /etc/firejail/newsboat.profile in any way before ?

**Reply by rusty-snake** [reply]
> Do i need to alter the /etc/firejail/newsboat.profile in any way before ?

No.

**Reply by githlp** [reply]
Well, since in newsboat.profile ${HOME}/.config/newsboat is noblacklist, what is then the problem with the newsboat-browser.sh ?

`#!/bin/sh`
`/usr/bin/firefox "$@" &`

**Reply by rusty-snake** [reply]
1. It is not executable (mounted `noexec`). Add `ignore noexec ${HOME}`.
2. You can not run Firefox in the newsboat sandbox.
3. Firefox can not IPC to a running instance for within the newsboat sandbox.
4. You script looks very useless.

**Reply by ghost** [reply]
I think you might want to edit ${HOME}/.config/newsboat/config. There's a `browser` command you can set there, e.g. `browser "xdg-open %u"`.
